lightning/src/offers/invoice_request.rs

```diff
@@ -1730,6 +1730,26 @@ mod tests {
 			Ok(_) => panic!("expected error"),
 			Err(e) => assert_eq!(e, Bolt12SemanticError::InvalidAmount),
 		}
+
+		let invoice_request = OfferBuilder::new(recipient_pubkey())
```

Comment:
What is meant by "However, the return behavior isn't what's expected; its a work in progress." in commit 73f643326dbf187a69423c5687c8e7b2542e6503?

Reply:
Ah, at that point, I was expecting the unit test to fail when creating an `InvoiceRequest` with an `Offer` that includes a currency and an amount, but I wasn’t sure why it didn’t. I should've updated the commit message after we clarified the behavior and I changed the test to succeed instead of fail.